packages/flutter_test/test/finders_test.dart

```diff
@@ -390,6 +395,391 @@ void main() {
+      await tester.pumpWidget(semanticsTree);
+
+      final FinderBase<SemanticsNode> finder = find.semantics.byLabel(RegExp('^label.*'));
+
+      expect(finder, findsExactly(5));
+      expect(finder.found.every((SemanticsNode node) => node.label.startsWith('label')), isTrue);
+    });
+
+    testWidgets('byValue finds nodes with matching value using String', (WidgetTester tester) async {
+      await tester.pumpWidget(semanticsTree);
+
+      final FinderBase<SemanticsNode> finder = find.semantics.byValue('value3');
+
+      expect(finder, findsOne);
+      expect(finder.found.first.value, 'value3');
+    });
+
+    testWidgets('byValue finds nodes with matching value using RegEx', (WidgetTester tester) async {
+      await tester.pumpWidget(semanticsTree);
+
+      final FinderBase<SemanticsNode> finder = find.semantics.byValue(RegExp('^value.*'));
+
+      expect(finder, findsExactly(5));
+      expect(finder.found.every((SemanticsNode node) => node.value.startsWith('value')), isTrue);
+    });
+
+    testWidgets('byHint finds nodes with matching hint using String', (WidgetTester tester) async {
+      await tester.pumpWidget(semanticsTree);
+
+      final FinderBase<SemanticsNode> finder = find.semantics.byHint('hint3');
+
+      expect(finder, findsOne);
+      expect(finder.found.first.hint, 'hint3');
+    });
+
+    testWidgets('byHint finds nodes with matching hint using RegEx', (WidgetTester tester) async {
+      await tester.pumpWidget(semanticsTree);
+
+      final FinderBase<SemanticsNode> finder = find.semantics.byHint(RegExp('^hint.*'));
+
+      expect(finder, findsExactly(5));
+      expect(finder.found.every((SemanticsNode node) => node.hint.startsWith('hint')), isTrue);
+    });
+
+    testWidgets('byAction finds nodes with matching action', (WidgetTester tester) async {
+      await tester.pumpWidget(semanticsTree);
+
+      final FinderBase<SemanticsNode> finder = find.semantics.byAction(SemanticsAction.copy);
+
+      expect(finder, findsExactly(3));
+    });
+
+    testWidgets('byAnyAction finds nodes with any matching actions', (WidgetTester tester) async {
+      await tester.pumpWidget(semanticsTree);
+
+      final FinderBase<SemanticsNode> finder = find.semantics.byAnyAction(<SemanticsAction>[
+        SemanticsAction.paste,
+        SemanticsAction.longPress,
+      ]);
+
+      expect(finder, findsExactly(4));
+    });
+
+    testWidgets('byFlag finds nodes with matching flag', (WidgetTester tester) async {
+      await tester.pumpWidget(semanticsTree);
+
+      final FinderBase<SemanticsNode> finder = find.semantics.byFlag(SemanticsFlag.isReadOnly);
+
+      expect(finder, findsExactly(3));
+    });
+
+    testWidgets('byAnyFlag finds nodes with any matching flag', (WidgetTester tester) async {
+      await tester.pumpWidget(semanticsTree);
+
+      final FinderBase<SemanticsNode> finder = find.semantics.byAnyFlag(<SemanticsFlag>[
```

Comment:
Yeah... Probably a good idea. I was going to argue that they're very fragile tests, but then I remembered that the _existing_ tests caught a couple bugs in the output, and you caught a couple more in review because of them. Proves my fragility argument to be a bad one. :P

Reply:
Messaging tests have been added!